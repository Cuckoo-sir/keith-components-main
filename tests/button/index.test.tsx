import { mount, render } from 'enzyme';
import Button from '@keith/button';
import React from 'react';

describe('Button', () => {
  it('renders correctly', () => {
    try {
      const html = render(<Button />);
      expect(html).toMatchSnapshot();
    } catch (error) {
      console.log(error);
    }
  });

  it('button size', () => {
    const wrapper1 = mount(<Button size="large">click me</Button>);
    expect(wrapper1.find('.btn-size-large').length).toBe(1);
    const wrapper2 = mount(<Button size="small">click me</Button>);
    expect(wrapper2.find('.btn-size-small').length).toBe(1);
    const wrapper3 = mount(<Button>click me</Button>);
    expect(wrapper3.find('.btn-size-middle').length).toBe(1);
  });
});
