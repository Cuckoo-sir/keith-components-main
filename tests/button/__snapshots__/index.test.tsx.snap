// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<button
  class="btn-shape-round btn-type-default btn-size-middle"
/>
`;
